radixdlt/src/main/java/com/radixdlt/consensus/validators/ValidationState.java

```diff
@@ -77,7 +77,8 @@ public boolean addSignature(ECPublicKey key, ECDSASignature signature) {
 	 * @return {@code true} if we have enough valid signatures to form a quorum,
 	 */
 	public boolean complete() {
-		return this.signedKeys.size() >= threshold(this.validKeys.size());
+		UInt256 currentPower = this.validatorSet.getPower(this.signedKeys.keySet());
+		return currentPower.compareTo(threshold(this.validatorSet.getTotalPower())) >= 0;
```

Comment:
Should precompute threshold, as this is relatively expensive, but fixed for given validator set.